### PR TITLE
Allow user-defined colors for single choice questions

### PR DESCRIPTION
**Pull request by okal**

### CODE REVIEW

**Comment by pld** [comment]
:+1:

**Comment by okal** [comment]
I think it should stay open until we conclude the discussion regarding the format, though.

File: src/hatti/map/viewby.cljs [hunk 1]
```diff
@@ -82,27 +46,29 @@
...
-        ans-s (preprocess-answers field raw-answers)
-        answer->count (frequencies (flatten ans-s))
-        sorted (cond
-                 (or (f/categorical? field)
-                     (f/text? field)
-                     (f/calculate? field))
+        preprocessed-answers (preprocess-answers field raw-answers)
+        answer->count (frequencies (flatten preprocessed-answers))
+        sorted-answers (cond
+                 (or (form-utils/categorical? field)
```
Comment (pld) [inline src/hatti/map/viewby.cljs]: not indented

Comment (okal) [inline src/hatti/map/viewby.cljs]: I wonder if we can we get the style check to catch these?

Comment (pld) [inline src/hatti/map/viewby.cljs]: Yeah, that would be great, I haven't come across a clj* style checker 
that does this, but I'll take another pass googling

Okal Otieno wrote:

Comment (okal) [inline src/hatti/map/viewby.cljs]: Or a pre-commit hook that does auto-indentation. But then we'd have to agree on a rule. Our indentation rules - across the team - seem to differ subtly.

**Comment by okal** [comment]
Do we need to update xlsform.org with this?

**Comment by okal** [comment]
I think the appearance attribute in the survey sheet tends to be on the ODK client side, rather than the data viz side, so I don't know.

**Pull request merged**